lib/speech/index.js

```diff
@@ -0,0 +1,490 @@
+/*!
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module speech
+ */
+
+'use strict';
+
+var arrify = require('arrify');
+var async = require('async');
+var extend = require('extend');
+var format = require('string-format-obj');
+var googleProtoFiles = require('google-proto-files');
+var fs = require('fs');
+var is = require('is');
+var nodeutil = require('util');
+var request = require('request');
+var through = require('through2');
+var duplexify = require('duplexify');
+
+/**
+ * @type {module:storage/file}
+ * @private
+ */
+var File = require('../storage/file.js');
+
+/**
+ * @type {module:common/grpc-service}
+ * @private
+ */
+var GrpcService = require('../common/grpc-service.js');
+
+/**
+ * @type {module:common/util}
+ * @private
+ */
+var util = require('../common/util.js');
+
+/**
+ * The [Cloud Speech API](https://cloud.google.com/speech/docs) enables easy
+ * integration of Google speech recognition technologies into developer
+ * applications. Send audio and receive a text transcription from the Cloud
+ * Speech API service.
+ *
+ * @constructor
+ * @alias module:speech
+ *
+ * @classdesc
+ * <p class="notice">
+ *   **This is an Alpha release of Google Cloud Speech.** This API is not
+ *   covered by any SLA or deprecation policy and may be subject to backward-
+ *   incompatible changes.
+ * </p>
+ *
+ * To learn more about the Speech API, see the
+ * [Getting Started guide](https://cloud.google.com/speech/docs/getting-started).
+ *
+ * @resource [Getting Started]{@link https://cloud.google.com/speech/docs/getting-started}
+ * @resource [Image Best Practices]{@link https://cloud.google.com/speech/docs/best-practices}
+ *
+ * @param {object} options - [Configuration object](#/docs).
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var speech = gcloud.speech();
+ */
+function Speech(options) {
+  if (!(this instanceof Speech)) {
+    options = util.normalizeArguments(this, options);
+    return new Speech(options);
+  }
+
+  var config = {
+    baseUrl: 'speech.googleapis.com',
+    projectIdRequired: false,
+    service: 'speech',
+    apiVersion: 'v1',
```

Comment:
Please use the v1beta1 version of the api rather than the alpha version (which unfortunately is named v1).  There are some breaking changes between the two, by the way.